### PR TITLE
Fix split and array_split with unordered indices supplied

### PR DESCRIPTION
**Pull request by takagi**
Follows #2814 .

`cupy.split` and `cupy.array_split` raise an error when they take unordered indices as `[4, 2]`. NumPy does not require indices ordered and, in such case, it return an empty array for a going-back index. I'm not sure it is meaningful, but it actually does. This PR fix the functions to follow NumPy's behavior.

```Python
>>> import numpy as np
>>> import cupy as cp
>>> np.split(np.array([0, 1, 2, 3, 4]), [4, 2])
[array([0, 1, 2, 3]), array([], dtype=int64), array([2, 3, 4])]
>>> cp.split(cp.array([0, 1, 2, 3, 4]), [4, 2])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "cupy/core/core.pyx", line 1341, in cupy.core.core.ndarray.__repr__
    return repr(self.get())
  File "cupy/core/core.pyx", line 1432, in cupy.core.core.ndarray.get
    a_cpu = numpy.empty(self._shape, dtype=self.dtype, order=order)
ValueError: negative dimensions are not allowed
```

### CODE REVIEW

File: cupy/core/_routines_manipulation.pyx [hunk 1]
```diff
@@ -435,7 +435,8 @@ cpdef array_split(ndarray ary, indices_or_sections, Py_ssize_t axis):
     if ary.size == 0:
         stride = 0
     for index in indices:
-        shape[axis] = index - prev
```
Comment (emcastillo) [inline cupy/core/_routines_manipulation.pyx]: I think that the way min & max are implemented in python can lead to multiple device synchronization issues here. 
Python could synchronize the device when reading the result of cupy.array.__lt__ and cupy.array.__gt__

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by takagi** [comment]
As #2814 is now sucessfully back-ported, would you make the back-ports of this PR as well?

**Comment by emcastillo** [comment]
Yes, I was waiting for that :)

**Comment by takagi** [comment]
Thanks!

**Pull request merged**